### PR TITLE
=str #16787 java cookbook

### PR DESCRIPTION
**Pull request by ktoso**
Missing a few recipes, I'm adding them now.

### CODE REVIEW

**Comment by ktoso** [comment]
Pushed fixup of mock `System.out.println()` which caused test failures. Ready to be reviewed now.

**Comment by ktoso** [comment]
refs #16787

**Comment by ktoso** [comment]
failure was pdf generation - file include was wrong in one spot `s/seq/list/`

File: akka-actor/src/main/scala/akka/util/ByteString.scala [hunk 1]
```diff
@@ -53,6 +54,14 @@ object ByteString {
   def fromArray(array: Array[Byte]): ByteString = apply(array)
 
   /**
+   * JAVA API
+   * Creates a new ByteString by copying an int array by converting from integral numbers to bytes.
+   */
+  @varargs
+  def fromInts(array: Int*): ByteString =
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/util/ByteString.scala]: this should not be changed here, but in release-2.3 branch

Comment (ktoso) [inline akka-actor/src/main/scala/akka/util/ByteString.scala]: ah, right - thanks!

File: akka-samples/akka-docs-java-lambda/src/test/java/docs/stream/cookbook/RecipeGlobalRateLimit.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+  }
+
+  final FlowMaterializer mat = ActorFlowMaterializer.create(system);
+
+  static
+  //#global-limiter-actor
+  public class Limiter extends AbstractActor {
+
+    public static class WantToPass { }
+    public static final WantToPass WANT_TO_PASS = new WantToPass();
```
Comment (patriknw) [inline akka-samples/akka-docs-java-lambda/src/test/java/docs/stream/cookbook/RecipeGlobalRateLimit.java]: Is WantToPass intended to be a singleton instance? Is it possible to use something more idiomatic, such as:

     public static class WantToPass {
         private WantToPass() {}
         public static final WantToPass INSTANCE = new WantToPass();
     }

Comment (ktoso) [inline akka-samples/akka-docs-java-lambda/src/test/java/docs/stream/cookbook/RecipeGlobalRateLimit.java]: Yes, I think this style caught my eye in some other spec. I'll move to INSTANCE

File: akka-stream/src/main/scala/akka/stream/javadsl/OperationAttributes.scala [hunk 3]
```diff
@@ -50,7 +50,9 @@ object OperationAttributes {
     private[akka] def asScala = scaladsl.OperationAttributes.dispatcher(dispatcher)
   }
 
-  private[akka] val none: OperationAttributes = new OperationAttributes {
+  def none() = _none
+
+  val _none: OperationAttributes = new OperationAttributes {
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/javadsl/OperationAttributes.scala]: why was this needed?

File: akka-samples/akka-docs-java-lambda/src/test/java/docs/stream/cookbook/RecipeGlobalRateLimit.java (hunk 2 above)
Comment (patriknw) [inline akka-samples/akka-docs-java-lambda/src/test/java/docs/stream/cookbook/RecipeGlobalRateLimit.java]: ok, thanks 
(several other places)

File: akka-stream/src/main/scala/akka/stream/javadsl/OperationAttributes.scala (hunk 3 above)
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/javadsl/OperationAttributes.scala]: Had to check up on this (didn't remember).
It's not really *needed*, but thought the reusing the none instance may be useful - shall I revert?

Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/javadsl/OperationAttributes.scala]: Had to check up on this - not really needed, I'll revert!

Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/javadsl/OperationAttributes.scala]: Oh, and about making it public - I think it's useful for end users to have access to none sometimes, in code like `if (x) none else name(...)` etc.

**Comment by ktoso** [comment]
Moved over to https://github.com/drewhk/akka/pull/22, applied Patriks comments.
